classes/condition.php

```diff
@@ -137,4 +136,67 @@ public function get_description($full, $not, info $info) {
     protected function get_debug_string() {
         return $this->coursecompleted ? '#' . 'True' : 'False';
     }
+
+    /**
+     * Checks whether this condition applies to user lists.
+     *
+     * @return bool True if this condition applies to user lists
+     */
+    public function is_applied_to_user_lists() {
+        // Course completions are assumed to be 'permanent', so they affect the
+        // display of user lists for activities.
+        return true;
+    }
+
+    /**
+     * Tests against a user list. Users who cannot access the activity due to
+     * availability restrictions will be removed from the list.
+     *
+     * @param array $users Array of userid => object
+     * @param bool $not If tree's parent indicates it's being checked negatively
+     * @param info $info Info about current context
+     * @param capability_checker $checker Capability checker
+     * @return array Filtered version of input array
+     */
+    public function filter_user_list(
+        array $users,
+        $not,
+        \core_availability\info $info,
+        \core_availability\capability_checker $checker) {
+
+        global $DB;
+
+        // If the array is empty already, just return it.
+        if (!$users) {
+            return $users;
+        }
+
+        $course = $info->get_course();
+        $cond = $this->coursecompleted ? 'NOT' : '';
+        $sql = "SELECT DISTINCT userid
+                  FROM {course_completions}
+                  WHERE timecompleted IS $cond NULL AND course = ?";
+        $compusers = $DB->get_records_sql($sql, [$course->id]);
```

Comment:
I did not test it but is it possible that a student does not have any records in the 'course_completions' table at all?
In this case the SQL for the "not completed" case would be wrong